statement.go

```diff
@@ -1011,5 +1026,10 @@ func (stmt *Stmt) ociStmtExecute(iters C.ub4, mode C.ub4) error {
 		mode,                // The mode: https://docs.oracle.com/cd/E11882_01/appdev.112/e10646/oci17msc001.htm#LNOCI17163
 	)
 
+	if (result != C.OCI_SUCCESS && result != C.OCI_SUCCESS_WITH_INFO) && stmt.cacheKey != "" {
+		// drop statement from cache for all errors when caching is enabled
+		stmt.releaseMode = C.ub4(C.OCI_STRLS_CACHE_DELETE)
```

Comment:
Could you check to see if casting to C.ub4 is needed here and in the other locations?